design/pedibus/override/templates/adults/webpage_adults.tpl

```diff
@@ -0,0 +1,130 @@
+{def $__TITLE = $node.name|wash()}
+{def $__LIMIT = ezini( 'IlPedibus', 'adults_page_limit', 'ilpedibus.ini' )}
+{def $__ADULTS = fetch(
+                        'content',
+                        'list',
+                        hash(
+                            'parent_node_id', $node.node_id,
+                            'class_identifier', 'adulto',
+                            'sort_by', array( 'priority', true() )
+                        )
+                )
+}
+
+{if $node.data_map.short_name.has_content}
+    {set $__TITLE = $node.data_map.short_name.content|wash()}
+{/if}
+<div class="container-fluid main_cage row_main_title adults">
+    <div class="row">
+        <div class="col-xs-12">
+            <h1>{$__TITLE}</h1>
+        </div>
+    </div>
+</div>
+
+{*{$__ADULTS|attribute(show)}*}
+
+{if $__ADULTS|count()}
+    {def $current_user = fetch( 'user', 'current_user' )}
+    {def $__LINES_FILTER = array()}
+    {if $current_user.is_logged_in}
+        {if $current_user.contentobject.class_identifier|eq("user_pedibus")}
+            {foreach $current_user.contentobject.data_map.lines_enable_to_read.content.relation_list as $__LINE_REF}
+                {set $__LINES_FILTER = $__LINES_FILTER|append($__LINE_REF.node_id)}
+            {/foreach}
+        {/if}
+    {/if}
+
+{*{$__LINES_FILTER|attribute(show)}*}
+
+{*<p>
+    A == {$current_user.is_logged_in|attribute(show)}
+</p>
+<p>
+    B == {$current_user.contentobject.name|attribute(show)}
+</p>*}
+
+    {cache-block subtree_expiry=$__ADULTS keys=concat($node.name|wash(),$node.node_id,"adults",$view_parameters.offset,$__LIMIT,$current_user.is_logged_in,$current_user.contentobject.name)}
```

Comment:
non capisco questo cache-block: si invalida al expire del subtree degli adulti
ma se aggiungo una disponibilità per un adulto che prima non l'aveva non viene invalidato
conviene usare lo smart view cache

Reply:
Sono inoltre sconsigliabili i cache-block in un full a meno che non ci sia un cache ttl=0

Reply:
Sono inoltre sconsigliabili i cache-block in un full a meno che non ci sia un cache ttl=0

Reply:
Su questo non sono d'accordo:
Il cache-block viene utilizzato per "cacheare" anche le immagini e altro senza doverle riprocessare un altra volta, anche per gli altri utenti il blocco, per cui, visto che mettendo un sistema di paginazione le uri cambiano, allora non ha nemmeno senso mettere un 
**{set-block scope=root variable=cache_ttl}0{/set-block}** all'inizio del template.
Ho in ogni modo cambiato la chiave del cache-block basandomi sul count degli oggetti, che dovrebbero variare se ne venissero creati di nuovi + l'eventuale offset della paginazione... in questo modo la cache viene creata per ogni pagina.
P.S: chiaramente con un limit di 100 (come impostato sul file ini) non so se si vedrà la differenza con o senza paginazione

Reply:
Non riesco a capire a cosa serve il cache-block.

Tu scrivi:
` {cache-block subtree_expiry=$__ADULTS keys=concat($node.name|wash(),$node.node_id,"adults",$view_parameters.offset,$__LIMIT,$current_user.is_logged_in,$current_user.contentobject.name)}`

_(dove tra l'altro `$__ADULTS ` è un'array di eZContentObjectTreeNode e invece dovrebbe essere un node_id o un url_alias vedi: https://doc.ez.no/eZ-Publish/Technical-manual/4.x/Reference/Template-functions/Miscellaneous/cache-block)_

quindi il cache-block viene invalidato quando viene aggiunto un nodo nel sottoalbero e varia a seconda della paginazione e del fatto che un utente sia loggato o meno: questo è il **comportamento standard della content cache** in una vista full (vedi https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/classes/eznodeviewfunctions.php#L313) con smart view cache configurato.

Reply:
E qui hai ragione, mi sa che mi sono confuso con un **keys**
Lo ho corretto con un $node.url_alias come **subtree_expiry**